src/commands.cpp

```diff
@@ -8250,4 +8252,108 @@ bool CommandBackupRemove::procresult(Result r)
     return r.wasErrorOrOK();
 }
 
+CommandGetBanners::CommandGetBanners(MegaClient* client)
+{
+    cmd("gban");
+
+    tag = client->reqtag;
+}
+
+bool CommandGetBanners::procresult(Result r)
+{
+    if (r.wasErrorOrOK())
+    {
+        client->app->getbanners_result(r.errorOrOK());
+        return true; // because parsing didn't fail
+    }
+
+    /*
+        {
+            "id": 2, ///The banner id
+            "t": "R2V0IFZlcmlmaWVk", ///Banner title
+            "d": "TWFrZSBpdCBlYXNpZXIgZm9yIHlvdXIgY29udGFjdHMgdG8gZmluZCB5b3Ugb24gTUVHQS4", ///Banner description.
+            "img": "Verified_image.png", ///Image name.
+            "l": "", ///URL
+            "bimg": "Verified_BG.png", ///background image name.
+            "dsp": "https://web-sandbox3.developers.mega.co.nz/images/mega/" ///Where to get the image.
+        }, {"id":3, ...}, ... ]
+    */
+
+    vector< tuple<int, string, string, string, string, string, string> > banners;
+
+    // loop array elements
+    while (client->json.enterobject())
+    {
+        int id = 0;
+        string title, description, img, url, bimg, dsp;
+        bool read = true;
+
+        // loop and read object members
+        while (read)
+        {
+            switch (client->json.getnameid())
+            {
+            case MAKENAMEID2('i', 'd'):
+                id = client->json.getint32();
+                break;
+
+            case MAKENAMEID1('t'):
+                client->json.storeobject(&title);
+                break;
+
+            case MAKENAMEID1('d'):
+                client->json.storeobject(&description);
+                break;
+
+            case MAKENAMEID3('i', 'm', 'g'):
+                client->json.storeobject(&img);
+                break;
+
+            case MAKENAMEID1('l'):
+                client->json.storeobject(&url);
+                break;
+
+            case MAKENAMEID4('b', 'i', 'm', 'g'):
+                client->json.storeobject(&bimg);
+                break;
+
+            case MAKENAMEID3('d', 's', 'p'):
+                client->json.storeobject(&dsp);
+                break;
+
+            case EOO:
+                read = false;
+                break;
+
+            default:
+                client->json.storeobject(); // skip unknown member
+            }
+        }
+
+        banners.emplace_back(make_tuple(id, move(title), move(description), move(img), move(url), move(bimg), move(dsp)));
+
+        client->json.leaveobject();
+    }
+
+    client->app->getbanners_result(move(banners));
+
+    return true;
+}
+
+CommandDismissBanner::CommandDismissBanner(MegaClient* client, int id)
+{
+    cmd("dban");
+    arg("id", id); // id of the Smart Banner
+    uint64_t msSinceEpoch = std::chrono::system_clock::now().time_since_epoch() / std::chrono::milliseconds(1);
```

Comment:
Well, at least to avoid using `time_t` because "encoding of calendar time in std::time_t is unspecified", even if _usually_ it is what we need. Do you want me to change it to use time()?